src/test/scala/users/storemanager/StoreManagerRepositoryTest.scala

```diff
@@ -0,0 +1,201 @@
+ *
+ * All Rights Reserved.
+ */
+
+package io.github.pervasivecats
+package users.storemanager
+
+import java.sql.DriverManager
+
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.duration.SECONDS
+
+import com.dimafeng.testcontainers.JdbcDatabaseContainer.CommonParams
+import com.dimafeng.testcontainers.PostgreSQLContainer
+import com.dimafeng.testcontainers.scalatest.TestContainerForAll
+import com.zaxxer.hikari.HikariConfig
+import com.zaxxer.hikari.HikariDataSource
+import eu.timepit.refined.auto.given
+import org.scalatest.EitherValues.given
+import org.scalatest.funspec.AnyFunSpec
+import org.scalatest.matchers.should.Matchers.*
+import org.testcontainers.utility.DockerImageName
+
+import users.user.valueobjects.{EncryptedPassword, PlainPassword, Username}
+import users.storemanager.Repository
+import users.storemanager.Repository.{StoreManagerAlreadyPresent, StoreManagerNotFound, OperationFailed}
+import users.storemanager.valueobjects.Store
+import users.storemanager.entities.StoreManager
+import users.ValidationError
+import users.user.services.PasswordAlgorithm
+
+class StoreManagerRepositoryTest extends AnyFunSpec with TestContainerForAll {
+
+  private val usernameString: String = "matteo"
+  private val storeID: Long = 10
+  private val newStoreID: Long = 99
+
+  private val username: Username = Username(usernameString).getOrElse(fail())
+  private val wrongUsername: Username = Username("nonmatteo").getOrElse(fail())
+  private val store: Store = Store(storeID).getOrElse(fail())
+  private val newStore: Store = Store(newStoreID).getOrElse(fail())
+
+  private val password: EncryptedPassword =
+    summon[PasswordAlgorithm].encrypt(PlainPassword("Password1!").getOrElse(fail())).getOrElse(fail())
+
+  val timeout: FiniteDuration = FiniteDuration(300, SECONDS)
+
+  val initScriptParam: CommonParams =
+    CommonParams(startupTimeout = timeout, connectTimeout = timeout, initScriptPath = Option("users.sql"))
+
+  override val containerDef: PostgreSQLContainer.Def = PostgreSQLContainer.Def(
+    dockerImageName = DockerImageName.parse("postgres:15.1"),
+    databaseName = "users",
+    username = "test",
+    password = "test",
+    commonJdbcParams = initScriptParam
+  )
+
+  @SuppressWarnings(Array("org.wartremover.warts.Var", "scalafix:DisableSyntax.var"))
+  private var repository: Option[Repository] = None
+
+  override def afterContainersStart(containers: Containers): Unit =
+    repository = Some(Repository.withPort(containers.container.getFirstMappedPort.intValue()))
+
+  describe("A PostgreSQL container") {
+    describe("when started") {
+      it("should stay connected") {
+        withContainers { pgContainer =>
+          Class.forName(pgContainer.driverClassName)
+          val connection = DriverManager.getConnection(pgContainer.jdbcUrl, pgContainer.username, pgContainer.password)
+          assert(!connection.isClosed)
+        }
+      }
+    }
+  }
```

Comment:
This test should be removed, it is not our duty to check that the "testcontainers" framework does indeed work